docs/operators/transform.md

```diff
@@ -0,0 +1,167 @@
+      <property>
+        <name>dt.operator.transform.expressionMap(isLeapYear)</name>
+        <value>{$.StartDate}.year().isLeap()</value>
+      </property>
+      <property>
+        <name>dt.operator.transform.expressionMap(Name)</name>
+        <value>org.apache.commons.lang3.StringUtils.joinWith(\" \", {$.FirstName}, {$.LastName})</value>
+      </property>
+```
+
+-   ***copyMatchingFields*** -   boolean
+    -   Specifies whether matching fields should be copied; here matching means the name and type of an input field is the same as the name and type of an output field. 
+        If the matching field appears in `expressionMap` then it ignores copy to output object.
+    -   Default Value = true.
+    
+#### Ports
+----------
+
+-   ***input*** -   Port for input tuples.
+    -   Mandatory input port
+    
+-   ***output***    -   Port for transformed output tuples.
+    -   Mandatory output port
+
+#### Attributes
+---------------
+-   ***Input port Attribute - input.TUPLE\_CLASS*** - Fully qualified class name and class should be Kryo serializable.
+    -   Mandatory attribute
+    -   Type of input tuple.
+
+-   ***Output port Attribute - output.TUPLE\_CLASS*** - Fully qualified class name and class should be Kryo serializable.
+    -   Mandatory attribute
+    -   Type of output tuple.
+    
+#### Application Example
+------------------------
+
+Please refer [Example](https://github.com/DataTorrent/examples/tree/master/tutorials/transform) for transform sample application.
+
+### Partitioning
+Being stateless, this operator can be partitioned using any of the built-in partitioners present in the Malhar library by setting a few properties as follows:
+
+#### Stateless partitioning
+Stateless partitioning will ensure that TransformOperator will will be partitioned right at the starting of the application and will remain partitioned throughout the lifetime of the DAG.
+TransformOperator can be stateless partitioned by adding following lines to properties.xml:
+
+```xml
+  <property>
+    <name>dt.operator.{OperatorName}.attr.PARTITIONER</name>
+    <value>com.datatorrent.common.partitioner.StatelessPartitioner:{N}/value>
+  </property>
+```
+
+where {OperatorName} is the name of the TransformOperator operator and
+      {N} is the number of static partitions.
+Above lines will partition TransformOperator statically {N} times. 
+
+#### Dynamic Partitioning
+Dynamic partitioning is a feature of Apex platform which changes the partition of the operator based on certain condition.
+TransformOperator can be dynamically partitioned using the below two partitioners:
+
+##### Throughput based
+Following code can be added to populateDAG(DAG dag, Configuration conf) method of application to dynamically partitioning TransformOperator:
+```java
+    StatelessThroughputBasedPartitioner<TransformOperator> partitioner = new StatelessThroughputBasedPartitioner<>();
+    partitioner.setCooldownMillis(conf.getLong(COOL_DOWN_MILLIS, 10000));
```

Comment:
Please don't put conf.getLong... Its confusing... Just put number to setColldownMillis and everything else.